presto-main/src/main/java/com/facebook/presto/server/PrestoServer.java

```diff
@@ -131,6 +131,24 @@ public void run()
             injector.getInstance(Announcer.class).start();
 
             log.info("======== SERVER STARTED ========");
+
+            ServerConfig serverConfig = injector.getInstance(ServerConfig.class);
+
+            if (serverConfig.isCoordinator() && serverConfig.isEnabledUIonSecondaryPort()) {
+                Bootstrap uIApp = new Bootstrap(new NodeModule(),
+                                                new DiscoveryModule(),
+                                                new JsonModule(),
+                                                new JaxrsModule(true),
+                                                new JmxModule(),
+                                                new JmxHttpModule(),
+                                                new JsonEventModule(),
+                                                new HttpEventModule(),
+                                                new ServerSecurityModule(),
+                                                new CoordinatorUIHttpServerModule(injector));
+                Injector uIInjector = uIApp.doNotInitializeLogging().initialize();
+
+                log.info("======== SECONDARY UI STARTED ========");
```

Comment:
We cannot include the UI module in the main Bootstrap since the HttpServerModule will bind the HttpServerInfo(which injects the HttpServerConfig from the singleton, and the HttpServerConfig can only have one HTTP port) into the singleton and HttpServerProvider takes that instance. And all these steps happened in the airlift repo. What I did in the UI module is rebinding the initialized components in the main injector into the Bootstrap for UI. The hacky way here to make the secondary port works is fetching the initialized HttpServerConfig, setting the http port from the ServerConfig and building a new HttpServerInfo and a new HttpServerProvider.